src/components/menu/_menu.js

```diff
@@ -177,61 +203,143 @@ function MenuDirective() {
   }
 }
 
-function MenuController($mdMenu, $attrs, $element, $scope) {
+function MenuController($mdMenu, $attrs, $element, $scope, $mdUtil, $timeout) {
+
   var menuContainer;
-  var ctrl = this;
+  var self = this;
   var triggerElement;
 
-  this.init = angular.bind(this, init);
-  this.open = angular.bind(this, openMenu);
-  this.close = angular.bind(this, closeMenu);
-
-  this.positionMode = angular.bind(this, positionMode);
-  this.offsets = angular.bind(this, offsets);
-
-  // Expose a open function to the child scope for html to use
-  $scope.$mdOpenMenu = this.open;
+  this.nestLevel = parseInt($attrs.mdNestLevel, 10) || 0;
 
   /**
    * Called by our linking fn to provide access to the menu-content
    * element removed during link
    */
-  function init(setMenuContainer) {
+  this.init = function init(setMenuContainer, opts) {
```

Comment:
Instead of putting the function onto `this`, they should be on `MenuController.prototype`; as it is written now, every menu instance will have duplicate function instances for all of `init`, `enableHoverListener`, etc.